### PR TITLE
CentOS 8

### PR DESCRIPTION
**Pull request by ramereth**
This adds support for CentOS 8. Some key differences between the previous
version of CentOS is the fact that IUS is no longer needed as it's intended that
CentOS Streams will provide other versions of PHP. Currently CentOS 8 only
supports php 7.2 so this includes that support.

### CODE REVIEW

**Comment by ramereth** [comment]
[osl-php](https://trello.com/c/qCKrGjLP/1542-osl-php)

File: spec/packages_spec.rb [hunk 1]
```diff
@@ -36,7 +42,12 @@
               expect { chef_run }.to_not raise_error
             end
             it do
-              php_pkg = version.to_f < 7 ? prefix : "mod_#{prefix}"
+              php_pkg =
```
Comment (Vancelot11) [inline spec/packages_spec.rb]: Would this be more readable as a case statement as you have above?
```ruby
php_pkg =
  case pltfrm
    when CENTOS_8
      'php'
    when CENTOS_7
      "mod_#{prefix}"
    else
      prefix
end
```

File: spec/packages_spec.rb [hunk 2]
```diff
@@ -22,7 +22,13 @@
         end
       end
       %w(5.3 5.6 7.1 7.2 7.3).each do |version|
```
Comment (Vancelot11) [inline spec/packages_spec.rb]:
```suggestion
      %w(5.3 5.6 7.1 7.2 7.3).each do |php_version|
```

File: test/integration/opcache-centos8/inspec/opcache-centos8_spec.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+['echo "<?php phpinfo(); ?>" | php', 'curl localhost'].each do |cmd|
```
Comment (Vancelot11) [inline test/integration/opcache-centos8/inspec/opcache-centos8_spec.rb]: Is this file needed?
Logic for opcache on Centos 8 is covered in centos-php72 and I don't see it referenced in the kitchen.yml

File: spec/packages_spec.rb (hunk 1 above)
Comment (ramereth) [reply spec/packages_spec.rb]: Kind of, I'll push up a change that's works properly but this exactly doesn't work.

**Comment by ramereth** [comment]
@Vancelot11 can you please re-review this?

**Comment by Vancelot11** [comment]
@ramereth Looks good to me

**Comment by ramereth** [comment]
!bump major *

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a major-level version bump to v5.0.0.  Have a nice day!
